env/env.go

```diff
@@ -48,27 +40,65 @@ func (p *Env) AddRootDir(baseDir string) error {
 	return nil
 }
 
-func (p *Env) AddEnvDir(envDir string) error {
+func (p *Env) isRootEnv(name string) bool {
+	for _, m := range p.RootDirMap {
+		for _, k := range m {
+			if k == name {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+type ActionType string
+
+const (
+	ActionTypePrepend     ActionType = "prepend"
+	ActionTypeAppend      ActionType = "append"
+	ActionTypeOverride    ActionType = "override"
```

Comment:
Maybe I missed it, but where do we use these three constants?
I see that we're calling `DefaultActionType` in many places, but it returns either `ActionTypeOverride` or `ActionTypePrependPath`.

Reply:
We use them here: https://github.com/buildpacks/lifecycle/pull/483/commits/d58ab15f71c49a6d063dc3d44975c2df73aedb71#diff-1e4c1fcda16b2f94c80249aa8e58bff941c8621fe28458d767fd3502c10c4f31R90-R101

Although that is a within package use and therefore they don't _need_ to be exported, it made sense to me to export them from a package design perspective, if we are exporting the other two. I.E if this package was a standalone entity what would make the most sense? Happy to hear the counter argument though.

I originally considered just passing the buildpack API into `AddEnvDir` and not exporting these at all, but decided on the current design because it is very explicit and maintains a separation of concerns. I could see `DefaultActionType` moving into a `buildpack` package that handles all buildpack API dependent logic in the future.

Reply:
I don't know why I ignored the usage in `AddEnvDir`. Probably it was too late :-D
I think the current implementation is great.
Thanks!